addons/stock_account/models/product.py

```diff
@@ -749,8 +749,13 @@ def _compute_average_price(self, qty_invoiced, qty_to_invoice, stock_moves, is_r
         candidates = stock_moves\
             .sudo()\
             .filtered(lambda m: is_returned == bool(m.origin_returned_move_id and sum(m.stock_valuation_layer_ids.mapped('quantity')) >= 0))\
-            .mapped('stock_valuation_layer_ids')\
-            .sorted()
+            .mapped('stock_valuation_layer_ids')
+
+        if len(candidates) > 1:
+            candidates = candidates.sorted()
```

Comment:
Thank you for your review. I've tested it with the key, and it still executes a query. In the test that I conducted with 100k stock moves, it resulted in 100k additional queries, which led to an increase of 30 more seconds with 100k.

Reply:
My comment is not very clear actually: In my proposal, you actually avoid the query done by https://github.com/odoo/odoo/blob/3f5cb9689bdda79e12e1e2e4c440cffaf7b79717/odoo/models.py#L6340 but because the prefetch is still broken at this step, it still generate one query to fetch `svl.create_date` by call.

Then you have to fix the prefetch **before** the `sorted`.

Reply:
@ryv-odoo Then, do I need to move it outside of the if condition?

Reply:
Just inverse the order of the conditions :thinking:

Reply:
@ryv-odoo I modified the code.